### ISSUE TITLE
Investigate keyboard shortcut for hide this page

### ISSUE THREAD

**Issue by owenatgov** [alphagov/govuk-design-system#2116]
## What
Investigate the use of a keyboard shortcut as part of the [hide this page work](https://github.com/alphagov/govuk-design-system/issues/1755).

## Why
To more easily allow users to use the button. The present hypothesis is that this will help screen reader and keyboard users who would otherwise struggle to reach the button if they needed to use it.

This is intended as an investigation with implementation as proof of concept to be potentially used in the research into hide this page being done within the next quarter https://github.com/alphagov/govuk-design-system/issues/2085

### Considerations
- The shortcut needs to be usable with only one hand for users with limited motor ability
- How will localised keyboards factor into the shortcut?
- What other apps might be using our proposed shortcut?
- With the current implementation proposal of 2 buttons in the markup of the page, we need to ensure this doesn't fire twice

## Who needs to work on this
@davidc-gds @owenatgov

- [x] Investigate possible options for keyboard shortcuts
- [x] Propose a shortcut to use
- [x] Implement into current prototype (https://github.com/alphagov/govuk-frontend/pull/2545)

**Comment by owenatgov** [comment]
We've made the decision to implement the `esc` key as a keyboard shortcut for now. This has been implemented on the draft proposals for both [govuk-frontend](https://github.com/alphagov/govuk-frontend/pull/2545) and [the design system](https://github.com/alphagov/govuk-design-system/pull/2079).

The reason we've gone this route is:

1. We want to test that a keyboard shortcut is actually useful to users before investing in the best choice for keyboard shortcut, therefore this is a means to test the functionality with users in our upcoming usability testing for the component.
2. Hannah Jump who worked on this in govuk made a good point about user expectation based on existing functionality:

>I think it’s important to look at other websites that people might be using. It’s quite dangerous if they’ve learnt that the escape key works on other websites, but then it doesn’t work on our website. I think women going through this are probably using other websites, including helplines, and the first thing they think of won’t be about child maintenance, they will be talking to other people first.
